certs/manager_test.go

```diff
@@ -268,41 +263,39 @@ func TestAddCertificate(t *testing.T) {
 	cert2Raw, _ := pem.Decode(cert2Pem)
 	cert2ID := HexSHA256(cert2Raw.Bytes)
 
-	tests := [...]struct {
+	emptyCertID := ""
+
+	withError := true
+	noError := false
+
+	testcases := []struct {
+		title  string
 		data   []byte
 		certID string
-		err    string
+		err    bool
 	}{
-		{[]byte(""), "", "Failed to decode certificate. It should be PEM encoded."},
-		{[]byte("-----BEGIN PRIVATE KEY-----\nYQ==\n-----END PRIVATE KEY-----"), "", "Failed to decode certificate. It should be PEM encoded."},
-		{[]byte("-----BEGIN CERTIFICATE-----\nYQ==\n-----END CERTIFICATE-----"), "", "asn1: syntax error"},
-		{certPem, certID, ""},
-		{combinedPemWrongPrivate, "", "tls: private key does not match public key"},
-		{combinedPem, cert2ID, ""},
-		{combinedPem, "", "Certificate with " + cert2ID + " id already exists"},
-		{pubPem, pubID, ""},
-		{expiredCertPem, "", "certificate is expired"},
+		{"empty cert", []byte(""), emptyCertID, withError},
```

Comment:
Can you please point me to the test case reproducing the issue?

Reply:
I think there should be a test case where you call `ValidateRequestCertificate` which fails before the fix and passes after the fix.

Reply:
It's hard to test this, as we'd need an integration test, or fake the whole http.Request (which people do not do).

I'd like to avoid rewriting this to be testable, as there is external testing in QA, which will result in a regression test.